### PR TITLE
feat(CORE-949): Add pipelines maintenance guide

### PR DESCRIPTION
**Pull request by MoonMoon1919**

### CODE REVIEW

File: _docs-sources/pipelines/maintain/updating.md [hunk 1]
```diff
@@ -1,6 +1,96 @@
...
+This guide assumes you have the following:
+- An AWS account with permissions to create the necessary resources
+- An [AWS Identity and Access Management](https://aws.amazon.com/iam/) (IAM) user or role with permissions to start pipelines deployments and update AWS Lambda functions
+- [AWS Command Line Interface](https://aws.amazon.com/cli/) (AWS CLI) installed on your local machine
+- [infrastructure-deployer](https://github.com/gruntwork-io/terraform-aws-ci/tree/main/modules/infrastructure-deployer) CLI tool installed locally
+- [aws-vault](https://www.github.com/99designs/aws-vault) installed locally for authenticating to AWS
+
+## Updating container images
+
+Gruntwork Pipelines uses two images - one for the[Deploy Runner](https://github.com/gruntwork-io/terraform-aws-ci/blob/main/modules/ecs-deploy-runner/docker/deploy-runner/Dockerfile) and one for [Kaniko](https://github.com/gruntwork-io/terraform-aws-ci/blob/main/modules/ecs-deploy-runner/docker/kaniko/Dockerfile). To update pipelines to the latest version, you must build and push new versions of each image.
```
Comment (arsci) [inline _docs-sources/pipelines/maintain/updating.md:20]:
```suggestion
Gruntwork Pipelines uses two images - one for the [Deploy Runner](https://github.com/gruntwork-io/terraform-aws-ci/blob/main/modules/ecs-deploy-runner/docker/deploy-runner/Dockerfile) and one for [Kaniko](https://github.com/gruntwork-io/terraform-aws-ci/blob/main/modules/ecs-deploy-runner/docker/kaniko/Dockerfile). To update pipelines to the latest version, you must build and push new versions of each image.
```

File: _docs-sources/pipelines/maintain/extending.md [hunk 2]
```diff
@@ -1,7 +1,119 @@
...
 
-Ug ug. Me type. You read. Me make words for big brain. Me use stick on rock. Me try hard. You see?
+Pipelines can be extended to support building and deploying application code in many repositories. This guide demonstrates how to update which repositories are allowed to submit requests to Pipelines to build docker images.
 
-Me have idea. Me tell you story. Long time ago, big mammoth roam land. Caveman hunt mammoth. Caveman win. Caveman happy.
 
-Me tired now. Me rest. You like words? Me make more later. Ug ug.
+## Adding a repository
+
+Pipelines has separate configurations for each type of job that be performed (e.g., building a docker image, running terraform plan, running terraform apply). An allow-list of repos and branches is defined for each job type, which can be updated to extend your usage of pipelines to additional application repositories.
```
Comment (arsci) [inline _docs-sources/pipelines/maintain/extending.md:8]:
```suggestion
Pipelines has separate configurations for each type of job that can be performed (e.g., building a docker image, running terraform plan, running terraform apply). An allow-list of repos and branches is defined for each job type, which can be updated to extend your usage of pipelines to additional application repositories.
```

File: _docs-sources/pipelines/maintain/extending.md [hunk 3]
```diff
@@ -1,7 +1,119 @@
...
+
+If you need to run a custom script in the `deploy-runner`, you must fork the image code, add an additional line to copy your script into directory designated by the `trigger_directory` argument. Then, you will need to rebuild the Docker image, push to ECR, then update your Pipelines deployment following the steps in [Updating Pipelines](./updating.md).
+
+## Adding permissions
+
+Pipelines executes in ECS tasks running in your AWS account(s). Each task (terraform planner, applier, docker builder, ami builder) has a distinct execution IAM role with only the permissions each task requires to complete successfully.
+
+If you are expanding your usage of AWS to include an AWS service you've never used before, you will need to grant each job sufficient permissions to access that service.
+For example, if you need to create an AWS DynamoDB Table using Pipelines for the first time, you would want to add (at a minimum) the ability to list and describe tables to the policy for the `planner` IAM role, and all permissions for DynamoDB to the IAM policy for the `terraform-applier` IAM role.
```
Comment (arsci) [inline _docs-sources/pipelines/maintain/extending.md:82]: `AWS DynamoDB`

I'm not sure how picky we usually are on this, but I think this is one of those services that uses the `Amazon` prefix and not `AWS`

File: _docs-sources/pipelines/maintain/updating.md [hunk 4]
```diff
@@ -1,6 +1,96 @@
...
+### Updating tag values (RefArch)
+
+Update `common.hcl` with new tag values for these images. The new tag value will be version of terraform-aws-ci that the images use. For example, if your newly created images are using the v0.52.1 release of terraform-aws-ci, update common.hcl to:
+
+```
+deploy_runner_container_image_tag = "v0.52.1"
+kaniko_container_image_tag = "v0.52.1"
+```
+
+Next, apply the ecs-deploy-runner module in each account.
```
Comment (arsci) [inline _docs-sources/pipelines/maintain/updating.md:66]:
```suggestion
Next, apply the ecs-deploy-runner module in each account:
```

File: _docs-sources/pipelines/maintain/extending.md (hunk 3 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/maintain/extending.md]: Fixed, thank you!

**Comment by arsci** [comment]
Actually I think you will need to regenerate so the changes from `_docs_sources` are reflected in the `docs` files (I only just thought of this after seeing some of the review changes from my PR not being reflected in the preview).

**Pull request merged**